fdbsyncd/fdbsyncd.cpp

```diff
@@ -45,7 +47,31 @@ int main(int argc, char **argv)
             if (sync.getRestartAssist()->isWarmStartInProgress())
             {
                 sync.getRestartAssist()->readTablesToMap();
+                
+                steady_clock::time_point starttime = steady_clock::now();
+                while (!sync.isIntfRestoreDone())
+                {
+                    duration<double> time_span =
+                        duration_cast<duration<double>>(steady_clock::now() - starttime);
+                    int pasttime = int(time_span.count());
+
+                    if (pasttime > INTF_RESTORE_MAX_WAIT_TIME)
+                    {
+                        SWSS_LOG_INFO("timed-out before all interface data was replayed to kernel!!!");

```

Comment:
System will proceed further. Some mac programming to kernel might fail because underlying interface is not yet created.

Reply:
If we could not restore interface, why we should proceed further and get into some limbo state that may or may not have critical issues.  I would suggest we abort to bring user's attention.

Reply:
Interfaces will be eventually restored. The only impact will be that warm-reboot might not be hitless and there will be traffic loss seen. Not sure if we need to go for full abort and impact everything and all traffic. Requesting @prsunny  to comment on this too

Reply:
"Some mac programming to kernel might fail because underlying interface is not yet created." so this condition will be recovered by someone later?  Again, if it is a critical condition, we should raise/abort so we don't get into limbo state.